vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt

```diff
@@ -64,24 +77,75 @@ class UserListViewModel @AssistedInject constructor(@Assisted initialState: User
         }
     }
 
+    private val identityServerListener = object : IdentityServiceListener {
+        override fun onIdentityServerChange() {
+            withState {
+                identityServerUsersSearch.accept(it.searchTerm)
+                setState {
+                    copy(
+                            configuredIdentityServer = cleanISURL(session.identityService().getCurrentIdentityServerUrl())
+                    )
+                }
+            }
+        }
+    }
+
     init {
         observeUsers()
+        setState {
+            copy(
+                    configuredIdentityServer = cleanISURL(session.identityService().getCurrentIdentityServerUrl())
+            )
+        }
+        session.identityService().addListener(identityServerListener)
+    }
+
+    private fun cleanISURL(url: String?): String? {
+        return url?.removePrefix("https://")
+    }
+
+    override fun onCleared() {
+        session.identityService().removeListener(identityServerListener)
+        super.onCleared()
     }
 
     override fun handle(action: UserListAction) {
         when (action) {
-            is UserListAction.SearchUsers -> handleSearchUsers(action.value)
-            is UserListAction.ClearSearchUsers -> handleClearSearchUsers()
-            is UserListAction.AddPendingSelection -> handleSelectUser(action)
-            is UserListAction.RemovePendingSelection -> handleRemoveSelectedUser(action)
+            is UserListAction.SearchUsers                -> handleSearchUsers(action.value)
+            is UserListAction.ClearSearchUsers           -> handleClearSearchUsers()
+            is UserListAction.AddPendingSelection        -> handleSelectUser(action)
+            is UserListAction.RemovePendingSelection     -> handleRemoveSelectedUser(action)
             UserListAction.ComputeMatrixToLinkForSharing -> handleShareMyMatrixToLink()
+            is UserListAction.UpdateUserConsent          -> handleISUpdateConsent(action)
         }.exhaustive
     }
 
+    private fun handleISUpdateConsent(action: UserListAction.UpdateUserConsent) {
+        viewModelScope.launch {
+            try {
+                session.identityService().setUserConsent(action.consent)
```

Comment:
if consent is given you should call `identityServerUsersSearch.accept(it.searchTerm)` here, no need to rely on onIdentityServerChange() and so to call onIdentityServerChange() when consent is given

Reply:
yes, but it make sense to notify no?

Reply:
Updated